Time scale for layers

4ian opened the pull request:
This refactor objects and layers to add custom time scale support for layers.
In particular `UpdateTime`/`updateTime` has been renamed `Update`/`update`. The whole scene is passed as an argument, avoiding some objects to retain a pointer to the scene which was hacky.
Runtime objects have a `GetElapsedTime` that is used to get the elapsed time for the object (that depends on the scene time scale and layer time scale).
A test game is included.

Actions/conditions/expressions are available to change the time scale for a layer (native and JS game engine support). 

This enables to "pause" the gameplay of a layer while still having other layers animations running (for example, an interface or a tutorial with animations displayed while the game is paused).

victorlevasseur commented:
The Travis CI PR build is failing:

```
/home/travis/build/4ian/GD/GDCpp/GDCpp/Runtime/RuntimeScene.cpp: In member function ‘void RuntimeScene::ManageObjectsAfterEvents()’:
/home/travis/build/4ian/GD/GDCpp/GDCpp/Runtime/RuntimeScene.cpp:328:36: error: invalid initialization of reference of type ‘RuntimeObjSPtr& {aka std::unique_ptr<RuntimeObject>&}’ from expression of type ‘RuntimeObject*’
     for (RuntimeObjSPtr & object : allObjects)
```

4ian commented:
Yes :( I'll fix that.

4ian commented:
@victorlevasseur I've fixed this branch (some changes related to shared_ptrs have been conflicting with a few changes I did) and added a few things.

In particular there is a small utility LinkRuntimeFiles.sh that can be useful for people wanting to develop the JS game engine (basically you export/preview a game and launch the script with the export dir as parameter to link all files to the Runtime directory. Much much faster to develop an extension or the game engine, rather than building and exporting at every change!). Works only on Linux/OS X though.
I think this should be added in some kind of documentation page but I don't know where.
